sw/device/lib/testing/test_framework/ottf_debugger.c

```diff
@@ -0,0 +1,36 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/json/command.h"
+
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/ottf_debugger.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf_commands.h"
+
+static status_t _ottf_debugger_loop(ujson_t *uj) {
+  LOG_INFO("Debugger started");
+  // FIXME implement getting out of the loop
+  while (true) {
+    test_command_t command;
+    TRY(ujson_deserialize_test_command_t(uj, &command));
+    if (command == kTestCommandExitDebugger)
+      return OK_STATUS();
+    status_t status = ujson_ottf_dispatch(uj, command);
+    if (status_err(status) == kUnimplemented) {
+      RESP_ERR(uj, status);
+    } else if (status_err(status) != kOk) {
+      return status;
```

Comment:
Yeah that's very much an open question. My thinking was that if something bad happens, the debugger loop stops but the caller of the debugger can decide what to do. On ottf abort, it just tries again so if the error is "not too bad", it will hopefully continue.